src/autocomplete.js

```diff
@@ -611,20 +619,33 @@ class CompletionProvider {
             // TODO add support for options.deleteSuffix
             if (!this.completions)
                 return false;
-            if (this.completions.filterText) {
+            
+            var replaceBefore = this.completions.filterText.length;
+            var replaceAfter = 0;
+            if (data.range && data.range.start.row === data.range.end.row) {
+                replaceBefore -= this.initialPosition.prefix.length;
```

Comment:
Are we sure this doesn't go out of bounds? The logic is very complicated at the moment, and I don't see any guarantees.